wolfcrypt/src/asn.c

```diff
@@ -5612,8 +5642,8 @@ static int DecodePolicyOID(char *out, word32 outSz, byte *in, word32 inSz)
             return ASN_PARSE_E;
         }
 
-        /* Validate total length (2 is the CERT_POLICY_OID+SEQ) */
-        if ((total_length + 2) != sz) {
+        /* Validate total length */
+        if (total_length > (sz - (int)idx)) {
```

Comment:
Are you saying the CERT_POLICY_OID + SEQ is not 2? Or that we didn't parse a byte?